Peaky/TestFailedException.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Peaky
+{
+    public class TestFailedException : Exception
```

Comment:
For what it's worth, we already have an `AssertionFailedException`. I like that this new exception class's name matches the others you've defined, though. Maybe we should obsolete `AssertionFailedException`?